### PR TITLE
[DOC] tours

### PR DESCRIPTION
**Pull request by bastien-ipb**

### CODE REVIEW

File: content/developer/reference/backend/testing.rst [hunk 1]
```diff
@@ -730,14 +731,113 @@ To start a tour from a python test, make the class inherit from
...
        # Optional verifications
 
+Writing an onboarding tour
+--------------------------
+
+The main purpose of the onboarding tours is to guide the user. They can be started with the
+checkbox in the user menu that will run all the onboarding tours in their sequence order.
+
+.. image:: testing/tour_user_menu.png
+   :align: center
```
Comment (AntoineVDV) [inline content/developer/reference/backend/testing.rst:744]: Actually I don't think that you need to add (and maintain) an image just for that:

```suggestion
The main purpose of the onboarding tours is to guide the user. They can all be started in their sequence order by toggling the :guilabel:`Onboarding` option in the user menu.
```

File: content/developer/reference/backend/testing.rst [hunk 2]
```diff
@@ -730,14 +731,113 @@ To start a tour from a python test, make the class inherit from
...
+
+    your_module
+    ├── ...
+    ├── data
+    |   └── your_tour.xml
+    ├── static
+    |   └── src
+    |       └── js
+    |           └── tours
+    |               └── your_tour.js
```
Comment (AntoineVDV) [inline content/developer/reference/backend/testing.rst:761]: Alternatively:
```suggestion
    ├── static/src/js/tours/your_tour.js
```

File: content/developer/reference/backend/testing.rst [hunk 3]
```diff
@@ -730,14 +731,113 @@ To start a tour from a python test, make the class inherit from
...
+         ],
+     },
+
+Javascript
+~~~~~~~~~~
+
+The javascript part is the same as for the test tour.
+
+.. seealso::
+   - :ref:`testing/javascript/test`
```
Comment (AntoineVDV) [inline content/developer/reference/backend/testing.rst:783]:
```suggestion
The javascript part is the same as for ref:`the test tour <testing/javascript/test>`.
```

File: content/developer/reference/backend/testing.rst [hunk 4]
```diff
@@ -730,14 +731,113 @@ To start a tour from a python test, make the class inherit from
...
+          </record>
+      </odoo>
+
+- **name**: Required, the name must be the same as the one in the
+  javascript registry.
+- **sequence**: Optional, will determine the order to execute the
+  onboarding tours, If no ``sequence``, has a default value of 1000.
+- **url**: Optional, the url where to start the tour. If no ``url``,
+  take the url from the registry.
+- **rainbow_man_message**: Optional, will show the message in the
```
Comment (AntoineVDV) [inline content/developer/reference/backend/testing.rst:807]:
```suggestion
- `name`: Required, the name must be the same as the one in the
  javascript registry.
- `sequence`: Optional, will determine the order to execute the
  onboarding tours, If no ``sequence``, has a default value of 1000.
- `url`: Optional, the url where to start the tour. If no ``url``,
  take the url from the registry.
- `rainbow_man_message`: Optional, will show the message in the
```

File: content/developer/reference/backend/testing.rst [hunk 5]
```diff
@@ -730,14 +731,113 @@ To start a tour from a python test, make the class inherit from
...
+              <field name="name">your_tour</field>
+              <field name="sequence">10</field>
+              <field name="rainbow_man_message">Congrats, that was a great tour</field>
+          </record>
+      </odoo>
+
+- **name**: Required, the name must be the same as the one in the
+  javascript registry.
+- **sequence**: Optional, will determine the order to execute the
+  onboarding tours, If no ``sequence``, has a default value of 1000.
```
Comment (AntoineVDV) [inline content/developer/reference/backend/testing.rst:804]:
```suggestion
- **sequence**: Optional; determines the order to execute the
  onboarding tours. Defaults to 1000.
```

File: content/developer/reference/backend/testing.rst [hunk 6]
```diff
@@ -730,14 +731,113 @@ To start a tour from a python test, make the class inherit from
...
+- **url**: Optional, the url where to start the tour. If no ``url``,
+  take the url from the registry.
+- **rainbow_man_message**: Optional, will show the message in the
+  rainbow man effect at the completion of the tour. If no ``rainbow_man_message``,
+  there is a default value. If you do not want a rainbow_man effect, then specify
+  the field and write nothing as value.
```
Comment (AntoineVDV) [inline content/developer/reference/backend/testing.rst:810]: Same as for `sequence`

File: content/developer/reference/backend/testing.rst [hunk 7]
```diff
@@ -730,14 +731,113 @@ To start a tour from a python test, make the class inherit from
...
    def test_your_test(self):
        # Optional Setup
-       self.start_tour("/web", 'your_module.your_tour_name', login="admin")
+       self.start_tour("/web", "your_tour_name", login="admin")
        # Optional verifications
 
+Writing an onboarding tour
+--------------------------
+
+The main purpose of the onboarding tours is to guide the user. They can be started with the
```
Comment (AntoineVDV) [inline content/developer/reference/backend/testing.rst:740]: Considering that you have a "Running onboardng tours" section, I would remove "The main purpose of the onboarding tours is to guide the user." completely and move the rest in that section.

File: content/developer/reference/backend/testing.rst [hunk 8]
```diff
@@ -730,14 +731,113 @@ To start a tour from a python test, make the class inherit from
...
+  rainbow man effect at the completion of the tour. If no ``rainbow_man_message``,
+  there is a default value. If you do not want a rainbow_man effect, then specify
+  the field and write nothing as value.
+
+Running onboarding tour
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Like said previously, you can activate the onboarding tour with the checkbox in user menu,
+but you can also run specific ones through the menu: ``Settings/Technical/User Interface/Tours``.
+In this view you'll get all the onboarding tours and you can execute them by clicking on `onboarding` or `testing`.
```
Comment (AntoineVDV) [inline content/developer/reference/backend/testing.rst:817]: With moving the intro here in mind:
```suggestion
You can run specific onboarding tours by going to the :menuselection:`Settings --> Technical --> User Interface --> Tours` and clicking on :guilabel:`Onboarding` or :guilabel:`Testing`.
```

Comment (AntoineVDV) [reply content/developer/reference/backend/testing.rst:817]: Given that this explanation is straightforward, I think that the second image should be removed too.

File: content/developer/reference/backend/testing.rst [hunk 9]
```diff
@@ -730,14 +731,113 @@ To start a tour from a python test, make the class inherit from
...
+  front of the user.
+
+Tour recorder
+~~~~~~~~~~~~~
+
+You can also create tour easily with the tour recorder, that will popup by clicking on `Record` on the
+onboarding tours view. When started, this tool will record all your interactions in Odoo.
+
+The created tours are listed in the onboarding tours view as `custom`. these tours can also
+be exported to javascript file, ready to be put in your module.
```
Comment (AntoineVDV) [inline content/developer/reference/backend/testing.rst:834]:
```suggestion
You can also create tours easily with the tour recorder. To do so, click on :guilabel:`Record` on the
onboarding tours view. When started, this tool will record all your interactions in Odoo.

The created tours are flagged in the onboarding tours view as **Custom**. These tours can also
be exported to a javascript file, ready to be put in your module.
```

File: content/developer/reference/backend/testing.rst [hunk 10]
```diff
@@ -730,14 +731,113 @@ To start a tour from a python test, make the class inherit from
...
+- **sequence**: Optional, will determine the order to execute the
+  onboarding tours, If no ``sequence``, has a default value of 1000.
+- **url**: Optional, the url where to start the tour. If no ``url``,
+  take the url from the registry.
+- **rainbow_man_message**: Optional, will show the message in the
+  rainbow man effect at the completion of the tour. If no ``rainbow_man_message``,
+  there is a default value. If you do not want a rainbow_man effect, then specify
+  the field and write nothing as value.
+
+Running onboarding tour
```
Comment (AntoineVDV) [inline content/developer/reference/backend/testing.rst:812]:
```suggestion
Running onboarding tours
```

File: content/developer/reference/backend/testing.rst [hunk 11]
```diff
@@ -782,22 +882,18 @@ debugger stops on the exception.
   - only works locally
   - only works if the test / tour can run correctly locally
 
-Run via browser
-***************
+Run test tour via browser
+*************************
```
Comment (AntoineVDV) [inline content/developer/reference/backend/testing.rst:886]: I think it's fine as "Run via browser" considering it's a nested section.

File: content/developer/reference/backend/testing.rst [hunk 12]
```diff
@@ -782,22 +882,18 @@ debugger stops on the exception.
...
 
-    odoo.startTour(tour_name);
+    odoo.startTour("tour_name");
 
 in the javascript console, or by enabling :ref:`tests mode
 <frontend/framework/tests_debug_mode>` by setting ``?debug=tests`` in
-the URL, then selecting **Start Tour** in the debug menu and picking a
-tour:
-
-.. image:: testing/tours.png
```
Comment (AntoineVDV) [inline content/developer/reference/backend/testing.rst:799]: Remove the image as well.

**Review by AntoineVDV — CHANGES_REQUESTED** [review]

File: content/developer/reference/backend/testing.rst [hunk 13]
```diff
@@ -730,14 +731,96 @@ To start a tour from a python test, make the class inherit from
...
+     'assets': {
+         'web.assets_backend': [
+             'your_module/static/src/js/tours/your_tour.js',
+         ],
+     },
+
+Javascript
+~~~~~~~~~~
+
+The javascript part is the same as for ref: `the test tour <testing/javascript/test>`.
```
Comment (AntoineVDV) [inline content/developer/reference/backend/testing.rst:770]: Broken ref
```suggestion
The javascript part is the same as for :ref:`the test tour <testing/javascript/test>`.
```

**Review by AntoineVDV — APPROVED** [review]
@robodoo delegate+

**Comment by bastien-ipb** [comment]
@robodoo r+

**Pull request closed without merging**